web/core/components/modules/links/create-update-modal.tsx

```diff
@@ -0,0 +1,152 @@
+"use client";
+
+import { FC, useEffect } from "react";
+import { Controller, useForm } from "react-hook-form";
+// plane types
+import type { ILinkDetails, ModuleLink } from "@plane/types";
+// plane ui
+import { Button, Input, ModalCore, setToast, TOAST_TYPE } from "@plane/ui";
+// helpers
+import { checkURLValidity } from "@/helpers/string.helper";
+
+type Props = {
+  createLink: (formData: ModuleLink) => Promise<void>;
+  data?: ILinkDetails | null;
+  isOpen: boolean;
+  handleClose: () => void;
+  updateLink: (formData: ModuleLink, linkId: string) => Promise<void>;
+};
+
+const defaultValues: ModuleLink = {
+  title: "",
+  url: "",
+};
+
+export const CreateUpdateModuleLinkModal: FC<Props> = (props) => {
+  const { isOpen, handleClose, createLink, updateLink, data } = props;
+  // form info
+  const {
+    formState: { errors, isSubmitting },
+    handleSubmit,
+    control,
+    reset,
+  } = useForm<ModuleLink>({
+    defaultValues,
+  });
+
+  const onClose = () => {
+    handleClose();
+    const timeoutId = setTimeout(() => {
+      reset(defaultValues);
+      clearTimeout(timeoutId);
+    }, 500);
+  };
+
+  const handleFormSubmit = async (formData: ModuleLink) => {
+    const payload = {
+      title: formData.title,
+      url: formData.url,
+    };
+
+    try {
+      if (!data) {
+        await createLink(payload);
+        setToast({
+          type: TOAST_TYPE.SUCCESS,
+          title: "Success!",
+          message: "Module link created successfully.",
+        });
+      } else {
+        await updateLink(payload, data.id);
+        setToast({
+          type: TOAST_TYPE.SUCCESS,
+          title: "Success!",
+          message: "Module link updated successfully.",
+        });
+      }
+      onClose();
+      reset(defaultValues);
```

Comment:
Wont calling `onClose` reset to `defaultValues`?

Reply:
Yes, it should. I'll correct this.